managed/resource_cluster_db_audit_export_config.go

```diff
@@ -0,0 +1,394 @@
+		if readInfoOK {
+			if asState == string(openapiclient.TASKACTIONSTATEENUM_SUCCEEDED) {
+				return nil
+			}
+			if asState == string(openapiclient.TASKACTIONSTATEENUM_FAILED) {
+				return fmt.Errorf("unable to associate cluster with db audit log configuration, operation failed")
+			}
+		} else {
+			return retry.RetryableError(errors.New("Unable to get check db audit log configuration cluster association: " + message))
+		}
+		return retry.RetryableError(errors.New("db audit log config is being associated to the cluster"))
+	})
+}
+
+func resourceAssociateDbAuditLogConfigClusterRead(ctx context.Context, accountId string, projectId string, clusterId string, configId string, apiClient *openapiclient.APIClient) (dbAuditExporterConfig DbAuditExporterConfig, readOK bool, err error) {
+	listDbAuditExporterConfigResp, _, err := apiClient.ClusterApi.ListDbAuditExporterConfig(ctx, accountId, projectId, clusterId).Execute()
+	if err != nil {
+		return dbAuditExporterConfig, false, err
+	}
+
+	dbAuditExporterConfig.AccountID.Value = accountId
+
+	for _, data := range listDbAuditExporterConfigResp.GetData() {
+		info := data.GetInfo()
+		spec := data.GetSpec()
+		if info.Id == configId {
+			dbAuditExporterConfig.ConfigID.Value = info.Id
+			dbAuditExporterConfig.ClusterID.Value = info.ClusterId
+			dbAuditExporterConfig.State.Value = string(info.State)
+			dbAuditExporterConfig.ExporterID.Value = spec.ExporterId
+
+			var logSettings LogSettings
+			logSettings.LogCatalog.Value = *spec.YsqlConfig.LogSettings.LogCatalog
+			logSettings.LogClient.Value = *spec.YsqlConfig.LogSettings.LogClient
+			logSettings.LogStatementOnce.Value = *spec.YsqlConfig.LogSettings.LogStatementOnce
+			logSettings.LogRelation.Value = *spec.YsqlConfig.LogSettings.LogRelation
+			logSettings.LogParameter.Value = *spec.YsqlConfig.LogSettings.LogParameter
+			logSettings.LogLevel.Value = string(*spec.YsqlConfig.LogSettings.LogLevel)
+
+			var statementClasses []types.String
+			for _, statementClass := range spec.GetYsqlConfig().StatementClasses {
+				statementClasses = append(statementClasses, types.String{Value: string(statementClass)})
+			}
+
+			var ysqlConfig YsqlConfig
+			ysqlConfig.StatementClasses = statementClasses
+			ysqlConfig.LogSettings = &logSettings
+
+			dbAuditExporterConfig.YsqlConfig = &ysqlConfig
+
+			return dbAuditExporterConfig, true, nil
+		}
+	}
+	return dbAuditExporterConfig, false, fmt.Errorf("unable to find db audit log cluster association with id %v for cluster %v", configId, clusterId)
+}
+
+// Read Db Audit log configuration for a cluster
+func (r resourceClusterDbAuditExportConfig) Read(ctx context.Context, req tfsdk.ReadResourceRequest, resp *tfsdk.ReadResourceResponse) {
+}
+
+// Update Db Audit log configuration for a cluster
+func (r resourceClusterDbAuditExportConfig) Update(ctx context.Context, req tfsdk.UpdateResourceRequest, resp *tfsdk.UpdateResourceResponse) {
```

Comment:
Can we not update anything ?

Reply:
Can you please elaborate here?

Reply:
Update not supported ?

Reply:
Update and deleted is supported but will add it in a separate PR for easy review